recipes/patchelf/all/test_package/conanfile.py

```diff
@@ -1,9 +1,15 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.build import cross_building
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
+    generators = "VirtualBuildEnv"
+    test_type = "explicit"
+
+    def build_requirements(self):
+        self.tool_requires(self.tested_reference_str)
 
     def test(self):
-        if not tools.cross_building(self, skip_x64_x86=True):
-            self.run("patchelf --version", run_environment=True)
+        if not cross_building(self, skip_x64_x86=True):
+            self.run("patchelf --version")
```

Comment:
Could we get a more meaningful test of the executable here? This tests that it's on the path but not that it works - I've had hit and miss experience with `--version`. Often it doesn't exercise whether the executable has been packaged correctly.

See:
* https://github.com/conan-io/conan-center-index/pull/16276#discussion_r1117892744
* https://github.com/conan-io/conan-center-index/issues/14120

Reply:
`patchelf` is self contained. There's no other files packaged than this single executable and the license so, from a packaging standpoint, `--version` is enough IMHO.